### PR TITLE
[ownership] But just for macOS

### PR DESCRIPTION
**Pull request by gottesmm**
I think we may be missing a pattern and thus are hitting large CFGs in i386. Just in case we run into a timeout issue, in this PR, I am just doing macOS.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
I did the same thing I did with Linux, namely that I made it so that macOS is the only platform that the stdlib gets the special treatment.

File: lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp [hunk 1]
```diff
@@ -2516,6 +2628,144 @@ bool AllocOptimize::tryToRemoveDeadAllocation() {
...
+          case TermKind::YieldInst: {
+            // These terminators can never be non-consuming uses of an owned
+            // value since we would be leaking the owned value no matter what
+            // we do. Given that, we can assume that what ever the
+            // non-consuming use actually was, must be before this
+            // instruction. So insert the destroy_value at the end of the
+            // block, before the terminator.
+            auto loc = RegularLocation::getAutoGeneratedLocation();
+            SILBuilderWithScope localBuilder(termInst);
+            localBuilder.createDestroyValue(loc, v);
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp:2749]: This is just `SILBuilderWithScope(termInst).createDestroyValue(loc, v)`

File: lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp [hunk 2]
```diff
@@ -2516,6 +2628,144 @@ bool AllocOptimize::tryToRemoveDeadAllocation() {
...
+      // of recomputing it there? Maybe an overload that lets us do this?
+      BasicBlockSet foundUseBlocks(v->getFunction());
+      for (auto *use : v->getUses()) {
+        auto *block = use->getParentBlock();
+        if (!foundUseBlocks.insert(block))
+          continue;
+        consumingUseBlocks.push_back(block);
+      }
+    }
+    findJointPostDominatingSet(
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp:2705]: fyi, the whole point of ValueLifetimeAnalysis is to do what you're doing manually here. That utility has been corrupted and is broken, so you shouldn't use it as-is, but I have a fixed, pure version on a branch that just does what you're doing manually and I'll put it up ASAP. Here you're putting destroys in successor blocks when they don't need to be, which seems like a bad idea but maybe was expedient.

**Review by atrick — APPROVED** [review]
I was little fuzzy on some of the PredictableMemOps logic, but overall everything looks ready to check in, with minor follow-up comments

**Comment by gottesmm** [comment]
I am actually doing the full thing here:

https://github.com/apple/swift/pull/35872.

File: lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp (hunk 2 above)
Comment (gottesmm) [reply lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp:2705]: SGTM

**Pull request closed without merging**